samples/sample_definition_and_criteria.rst

```diff
@@ -39,36 +47,42 @@ Sample Criteria
             regex:
               - ".*Waited and joined with 3 threads"
 
-  **Guidelines for Samples:**
-    * Minimize the use of ``platform_allow`` and architecture filters.
-    * Do not mark the test as build only unless necessary.
-    * Any test case-specific configuration options are added using ``extra_args`` or
-      ``extra_configs`` options in the YAML file. The `prj.conf` file should have the in-common
-      configuration options.
-    * The sample itself can be evaluated using multiple configurations in the sample's YAML file.
-      For example, the sample is used to run with different logging modes using multiple scenarios
-      in ``samples/subsys/logging/syst``.
-    * Sample output can be validated leveraging the ``harness_config`` regex option,
-      wherever applicable.
-
-3. Samples should target multiple platforms and architectures.
-++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++
-  * Minimize the use of ``platform_allow`` and architecture filters as it limits the scope
-    of testing to the mentioned platforms and architectures.
-    Reference: :ref:`twister_script`
+  * Do not mark the test as build only. A sample should be able to build and
+    run on the documented platforms. Use the harness option to skip the
+    execution. Twister only attempts to flash and execute the build binary if
+    the harness is set to ztest or console.
+  * The default configuration for the sample should be in the ``prj.conf``
+    file and should be operational on all supported platforms. Do not rely on the
+    ``extra_args`` or ``extra_configs`` options in the YAML file to build the
+    sample.
+  * The tests should verify the default configuration of the sample and any
+    optional features documented in the sample's README file. Sample should
+    never be used to test functionality of the subsystem or module the sample
+    belongs to.
+  * Any documented configuration options related to the sample and its
+    operation can be verified using ``extra_args`` or
+    ``extra_configs`` options in the YAML file. The `prj.conf` file should have the
+    base configuration options.
+  * Sample output can be validated leveraging the ``harness_config`` regex option,
+    wherever applicable.
+  * Use ``platform_allow`` to limit test to the platforms the sample was
+    verified on. Those platforms should be listed in the sample's README file.
   * Make use of ``integration_platforms`` to limit the scope when there are timing or
-    resource constraints.
+    resource constraints. Ideally, one platform should be enough to verify the
+    sample when changes are submitted to the Zephyr tree via a pull-request.
   * Make the sample as generic as possible. Avoid making a sample platform specific unless it is
     for particular hardware.
 
-4. A sample should provide sufficient coverage of a subsystem, feature, or module.
+3. A sample should provide sufficient coverage of a subsystem, feature, or module.
```

Comment:
FWIW I have mixed feelings about this section as it contains a lot of redundant information (if we people don't get it by now that samples must not be unit tests!) and the wording might be confusing: we ask for "sufficient coverage of a subsytesm" when earlier we advise to not use samples to test subsystems.
No immediate action needed though as it is OK enough as-is and can be improved in later PRs.

Reply:
coverage as in 'feature coverage', try to demonstrate as much as possible of a subsystem to be usable and submit thin content leaving much behind.

Reply:
yup, I know that's what this is *supposed* to mean, hence why I said I think it's *confusing* as other people not aware of the "gating" we're trying to establish might interpret "coverage" in a different way. Like I said, will likley tackle this in a follow-up PR but I guess where my head is at is rewording/summarizing this along the lines of "A sample should demonstrate an actual use case" or something in that vein.